test/stress_tests/run_application_stress_tests.sh

```diff
@@ -0,0 +1,118 @@
+        echo "Current Directory:" $(pwd)
+        cat application_cluster_template.yaml |
+            sed -e "
+                s/<<<CLUSTER_NAME>>>/rllib-impala-testing/;
+                s/<<<RAY_VERSION>>>/$RAY_VERSION/;
+                s/<<<HEAD_TYPE>>>/g3.16xlarge/;
+                s/<<<WORKER_TYPE>>>/m5.24xlarge/;
+                s/<<<MIN_WORKERS>>>/5/;
+                s/<<<MAX_WORKERS>>>/5/;
+                s/<<<PYTHON_VERSION>>>/$PYTHON_VERSION/;
+                s/<<<WHEEL_STR>>>/$WHEEL_STR/;" > $CLUSTER
+
+        echo "Try running IMPALA stress test."
+        {
+            RLLIB_DIR=../../python/ray/rllib/
+            ray up -y $CLUSTER
+            ray rsync_up $CLUSTER $RLLIB_DIR/tuned_examples/ tuned_examples/
+            sleep 1
+            ray exec $CLUSTER "
+                rllib train -f tuned_examples/atari-impala-large.yaml --redis-address='localhost:6379' --queue-trials"
+
+            echo "PASS: IMPALA Test for" $PYTHON_VERSION >> $RESULT_FILE
+        } || echo "FAIL: IMPALA Test for" $PYTHON_VERSION >> $RESULT_FILE
+
+        # Tear down cluster.
+        if [ "$DEBUG_MODE" = "" ]; then
+            ray down -y $CLUSTER
+            rm $CLUSTER
+        else
+            echo "Not tearing down cluster" $CLUSTER
+        fi
+    popd
+}
+
+
+test_sgd(){
+    local PYTHON_VERSION=$1
+    local WHEEL_STR=$(_find_wheel_str $PYTHON_VERSION)
+
+    pushd "$ROOT_DIR"
+        local CLUSTER="sgd.yaml"
+        echo "Creating SGD cluster YAML from template."
+
+        cat application_cluster_template.yaml |
+            sed -e "
+                s/<<<CLUSTER_NAME>>>/sgd-testing/;
+                s/<<<RAY_VERSION>>>/$RAY_VERSION/;
+                s/<<<HEAD_TYPE>>>/g3.16xlarge/;
+                s/<<<WORKER_TYPE>>>/g3.16xlarge/;
+                s/<<<MIN_WORKERS>>>/3/;
+                s/<<<MAX_WORKERS>>>/3/;
+                s/<<<PYTHON_VERSION>>>/$PYTHON_VERSION/;
+                s/<<<WHEEL_STR>>>/$WHEEL_STR/;" > $CLUSTER
+
+        echo "Try running SGD stress test."
+        {
+            SGD_DIR=$ROOT_DIR/../../python/ray/experimental/sgd/
+            ray up -y $CLUSTER
+            # TODO: fix submit so that args work
+            ray rsync_up $CLUSTER $SGD_DIR/mnist_example.py mnist_example.py
+            sleep 1
+            ray exec $CLUSTER "
+                python mnist_example.py --redis-address=localhost:6379 --num-iters=2000 --num-workers=8 --devices-per-worker=2 --gpu"
+
+            echo "PASS: SGD Test for" $PYTHON_VERSION >> $RESULT_FILE
+        } || echo "FAIL: SGD Test for" $PYTHON_VERSION >> $RESULT_FILE
+
+        # Tear down cluster.
+        if [ "$DEBUG_MODE" = "" ]; then
+            ray down -y $CLUSTER
+            rm $CLUSTER
+        else
+            echo "Not tearing down cluster" $CLUSTER
+        fi
+    popd
+}
+
+# RUN TESTS
+for PYTHON_VERSION in "p27" "p36"
+do
+    test_impala $PYTHON_VERSION
+    test_sgd $PYTHON_VERSION
```

Comment:
Should launch everything in parallel :)

Reply:
You could do it like this I think:

test_impala $PYTHON_VERSION &
impala_pid=$?
test_sgd  ...
sgd_pid=$?

wait $impala_pid $sgd_pid

Reply:
OK for this first cut I'm not going to launch in parallel because it makes it really hard to iterate/stop launches, but we can do this after the automation workflow is smooth.